### PR TITLE
feat: added copy feature

### LINKED ISSUE #173: [FEATURE] Add a copy button.
### Description

Hello:)
I think it is a good idea to add a copy badge icon.

### Screenshots

image

### Additional information

I would like to work on this

### PR DESCRIPTION
**Pull request by Bashamega**
## Fixes Issue

closes #173 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

### CODE REVIEW

File: src/app/account/repo/checks/[id]/form.js [hunk 1]
```diff
@@ -19,6 +21,21 @@ export default function Form({ id }) {
...
 
 export function FormBadge({ src }) {
+  const [copy, setCoppy] = useState(false);
+  const copyHandle = () => {
+    const url = `![HealthCheck](${src})`;
+
+    // Use the Clipboard API to copy the URL
+    navigator.clipboard
+      .writeText(url)
+      .then(() => {
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:31]: Please can you change this to `async/await` so it is consistent coding style

File: src/app/account/repo/checks/[id]/form.js [hunk 2]
```diff
@@ -19,6 +21,21 @@ export default function Form({ id }) {
...
+        console.log("Copied to clipboard:", url);
+        setCoppy(true);
+      })
+      .catch((err) => {
+        console.error("Failed to copy: ", err);
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:36]: I don't think we need the console logs, but can display an error message (simialr to the `Copied!` but in red

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 

Please can you add a screenshot for desktop and mobile view.

**Comment by eddiejaoude** [comment]
@Bashamega you requested a review but there are no changes since my last review - did you forget to push?

**Comment by Bashamega** [comment]
> @Bashamega you requested a review but there are no changes since my last review - did you forget to push?
> 
> note: please don't mark inline discussions as resolved, as it makes it more difficult to review

I am very sorry. I have forgot to push the changes. LOL!

File: src/app/account/repo/checks/[id]/form.js [hunk 3]
```diff
@@ -27,12 +35,13 @@ export function FormBadge({ src }) {
       >
         Add badge to your Repo&lsquo;s README to show the latest check status
       </label>
-      <div className="mt-2 flex rounded-md shadow-sm">
+      <div className="mt-2  rounded-md shadow-sm">
         <div className="relative flex flex-grow items-stretch focus-within:z-10">
-          <div className="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3">
-            <CheckBadgeIcon
+          <div className=" absolute inset-y-0 left-0 flex items-center pl-3 cursor-pointer">
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:40]: We have an `Input` component that has a `prepend` I was thinking that could be used? With some small tweaks for the onclick.

https://github.com/EddieHubCommunity/HealthCheck/blob/main/src/components/forms/Input.js#L37

**Comment by Bashamega** [comment]
Hello @eddiejaoude I have updated the form to use the Input component and made the input component more customizable

File: src/components/forms/Input.js [hunk 4]
```diff
@@ -58,8 +62,8 @@ export default function Input({
   }
 
   return (
-    <div className="col-span-full">
-      <label htmlFor={id} className="block text-sm font-medium leading-6">
+    <div className={"col-span-full " + className}>
```
Comment (eddiejaoude) [inline src/components/forms/Input.js:65]: please use `classNames` for this also

Comment (Bashamega) [reply src/components/forms/Input.js]: I have integrate the usage of classnames.

File: src/components/forms/Input.js [hunk 5]
```diff
@@ -58,8 +62,14 @@ export default function Input({
   }
 
   return (
-    <div className="col-span-full">
-      <label htmlFor={id} className="block text-sm font-medium leading-6">
+    <div className={classNames("col-span-full ", className)}>
```
Comment (eddiejaoude) [inline src/components/forms/Input.js:65]: Extra space not needed here between strings, the function will do it for us
```suggestion
    <div className={classNames("col-span-full", className)}>
```

File: src/app/account/repo/checks/[id]/form.js [hunk 6]
```diff
@@ -19,31 +21,38 @@ export default function Form({ id }) {
...
             name="badge"
             type="text"
             value={`![HealthCheck](${src})`}
-            className="block w-full rounded-md border-0 py-1.5 pl-10 text-gray-900 ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
+            className="w-full -z-30"
+            inputClassName="pl-10 py-1.5"
+            labelClassName="hidden"
           />
         </div>
+        {copy && <p className=" text-green-400 mt-2">Copied!</p>}
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:55]: Extra space not needed here

```suggestion
        {copy && <p className="text-green-400 mt-2">Copied!</p>}
```

File: src/app/account/repo/checks/[id]/form.js [hunk 7]
```diff
@@ -19,31 +21,38 @@ export default function Form({ id }) {
...
-      <div className="mt-2 flex rounded-md shadow-sm">
+      <div className="mt-2  rounded-md shadow-sm">
         <div className="relative flex flex-grow items-stretch focus-within:z-10">
-          <div className="pointer-events-none absolute inset-y-0 left-0 flex items-center pl-3">
-            <CheckBadgeIcon
+          <div className=" absolute inset-y-0 left-0 flex items-center pl-3 cursor-pointer">
+            <ClipboardDocumentCheckIcon
               aria-hidden="true"
-              className="h-5 w-5 text-gray-400"
+              className={`h-5 w-5 ${copy ? "text-green-400 " : "text-gray-500 "}  `}
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:40]: Best to be consistent and use `classNames` function here to concatenate class strings

File: src/app/account/repo/checks/[id]/form.js (hunk 3 above)
Comment (eddiejaoude) [reply src/app/account/repo/checks/[id]/form.js:37]: Extra space not required 
```suggestion
          <div className="absolute inset-y-0 left-0 flex items-center pl-3 cursor-pointer">
```

File: src/app/account/repo/checks/[id]/form.js [hunk 8]
```diff
@@ -19,31 +21,38 @@ export default function Form({ id }) {
 }
 
 export function FormBadge({ src }) {
+  const [copy, setCoppy] = useState(false);
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:24]: Too many `p`s

```suggestion
  const [copy, setCopy] = useState(false);
```

Comment (Bashamega) [reply src/app/account/repo/checks/[id]/form.js:24]: nice

**Comment by Bashamega** [comment]
I have fixed the problems :)

File: src/components/forms/Input.js (hunk 4 above)
Comment (eddiejaoude) [reply src/components/forms/Input.js]: Thank you

File: src/app/account/repo/checks/[id]/form.js [hunk 9]
```diff
@@ -19,31 +22,41 @@ export default function Form({ id }) {
 }
 
 export function FormBadge({ src }) {
+  const [copy, setCopy] = useState(false);
+  const copyHandle = async () => {
+    const url = `![HealthCheck](${src})`;
+    await navigator.clipboard.writeText(url);
+    setCoppy(true);
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:29]: This is spelt incorrectly and not matching the set function `setCopy`

```suggestion
    setCopy(true);
```

File: src/app/account/repo/checks/[id]/form.js [hunk 10]
```diff
@@ -3,11 +3,14 @@
 import {
   DocumentDuplicateIcon,
   CheckBadgeIcon,
```
Comment (eddiejaoude) [inline src/app/account/repo/checks/[id]/form.js:5]: I think `CheckBadgeIcon` is no longer used.

Don't worry about fixing it, I will merge to a temporary branch and fix now, so we can get this feature merged

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**